tests/spec/corpus.rs

```diff
@@ -0,0 +1,226 @@
+            bson::encode_document(
+                &mut native_to_bson_json_to_native_cej,
+                json_to_native_cej.as_document().unwrap(),
+            )
+            .unwrap();
+
+            if test.bson_type != "0x13" {
+                assert_eq!(
+                    hex::encode(native_to_bson_json_to_native_cej).to_lowercase(),
+                    valid.canonical_bson.to_lowercase(),
+                    "{}",
+                    description,
+                );
+            }
+        }
+
+        // native_to_bson( bson_to_native(dB) ) = cB
+
+        if let Some(db) = valid.degenerate_bson {
+            let bson_to_native_db =
+                bson::decode_document(&mut hex::decode(&db).expect(&description).as_slice())
+                    .expect(&description);
+
+            let mut native_to_bson_bson_to_native_db = Vec::new();
+            bson::encode_document(&mut native_to_bson_bson_to_native_db, &bson_to_native_db)
+                .unwrap();
+
+            assert_eq!(
+                hex::encode(native_to_bson_bson_to_native_db).to_lowercase(),
+                valid.canonical_bson.to_lowercase(),
+                "{}",
+                description,
+            );
+        }
+
+        if let Some(ref degenerate_extjson) = valid.degenerate_extjson {
+            let dej: serde_json::Value =
+                serde_json::from_str(degenerate_extjson).expect(&description);
+
+            let json_to_native_dej: Bson = dej.clone().into();
+
+            // native_to_canonical_extended_json( json_to_native(dEJ) ) = cEJ
+
+            let native_to_canonical_extended_json_json_to_native_dej =
+                json_to_native_dej.clone().into_canonical_extjson();
+
+            if test.bson_type != "0x13" {
+                assert_eq!(
+                    native_to_canonical_extended_json_json_to_native_dej, cej,
+                    "{}",
+                    description,
+                );
+            }
+
+            // native_to_bson( json_to_native(dEJ) ) = cB (unless lossy)
+
+            if valid.lossy != Some(true) {
+                let mut native_to_bson_json_to_native_dej = Vec::new();
+                bson::encode_document(
+                    &mut native_to_bson_json_to_native_dej,
+                    json_to_native_dej.as_document().unwrap(),
+                )
+                .unwrap();
+
+                if test.bson_type != "0x13" {
+                    assert_eq!(
+                        hex::encode(native_to_bson_json_to_native_dej).to_lowercase(),
+                        valid.canonical_bson.to_lowercase(),
+                        "{}",
+                        description,
+                    );
+                }
+            }
+        }
+
+        // native_to_relaxed_extended_json( json_to_native(rEJ) ) = rEJ
+
+        if let Some(ref rej) = valid.relaxed_extjson {
+            let json_to_native_rej: Bson = cej.clone().into();
```

Comment:
this should be `rej.clone().into()`

Reply:
Oh shoot, good catch! I knew that these long names were worth it